build/document-extractor.js

```diff
@@ -322,22 +333,50 @@ function _addSectionProse($) {
   let titleAsText = null;
   let isH3 = false;
 
-  // Maybe this should check that the h2 is first??
+  const flaws = [];
+
+  // The way this works...
+  // Given a section of HTML, try to extract a id, title,
+
+  let h2found = false;
   const h2s = $.find("h2");
-  if (h2s.length === 1) {
-    id = h2s.attr("id");
-    title = h2s.html();
-    titleAsText = h2s.text();
-    h2s.remove();
-  } else {
+  for (const i of [...Array(h2s.length).keys()]) {
+    if (i) {
+      // Excess!
+      flaws.push(
+        `Excess <h2> tag that is NOT at root-level (id='${h2s
+          .eq(i)
+          .attr("id")}', text='${h2s.eq(i).text()}')`
+      );
+    } else {
+      // First element
+      id = h2s.eq(i).attr("id");
+      title = h2s.eq(i).html();
+      titleAsText = h2s.eq(i).text();
+      h2s.eq(i).remove();
+    }
+    h2found = true;
+  }
+
+  // If there was no <h2>, look through all the <h3>s.
+  if (!h2found) {
     const h3s = $.find("h3");
-    if (h3s.length === 1) {
-      id = h3s.attr("id");
-      title = h3s.html();
-      titleAsText = h3s.text();
-      if (id && title) {
-        isH3 = true;
-        h3s.remove();
+    for (const i of [...Array(h3s.length).keys()]) {
```

Comment:
That feels like being ripe for refactoring down the road.
The logic is quite similar to the one used for `h2`